app/commands/cohort/join.rb

```diff
@@ -0,0 +1,15 @@
+class Cohort::Join
+  include Mandate
+
+  initialize_with :user, :cohort, :introduction
+
+  def call
+    # TODO: make this safe against race issues
+    user.cohort_memberships.create!(cohort:, status:, introduction:)
+  rescue ActiveRecord::RecordNotUnique
+    # This is fine
```

Comment:
You might want to lookup + return the correct membership back from here so that this always returns a membership